File: alica-supplementary/plan-conversion-tool/src/main/java/com/rapyutarobotics/alica/factories/RuntimeConditionFactory.java
package com.rapyutarobotics.alica.factories;

import de.unikassel.vs.alica.planDesigner.alicamodel.RuntimeCondition;
import org.w3c.dom.Element;

public class RuntimeConditionFactory extends Factory {

    public static RuntimeCondition create(Element runtimeConditionNode) {
        RuntimeCondition runtimeCondition = new RuntimeCondition();
        ConditionFactory.fillConditon(runtimeConditionNode, runtimeCondition);
        return runtimeCondition;
    }
}

File: alica_engine/src/engine/modelmanagement/factories/RuntimeConditionFactory.cpp
#include "engine/modelmanagement/factories/RuntimeConditionFactory.h"
#include "engine/modelmanagement/factories/ConditionFactory.h"
#include "engine/modelmanagement/Strings.h"
#include "engine/model/RuntimeCondition.h"

namespace alica
{
    RuntimeCondition* RuntimeConditionFactory::create(const YAML::Node& runtimeConditionNode, AbstractPlan* plan)
    {
        RuntimeCondition* runtimeCondition = new RuntimeCondition();
        ConditionFactory::fillCondition(runtimeConditionNode, runtimeCondition, plan);
        return runtimeCondition;
    }
} // namespace alica
